src/frontend/frontend/console.js

```diff
@@ -46,214 +46,113 @@ angular.module('frontend-app')
-        stack_trace(contents);
-      }
-      else if(/heap-buffer-(over|under)flow /.test(contents[1])) { // heap buffer overflow
-        var heapErrorType = /(overflow|underflow)/.exec(contents[1])[1];
-        self._write(sprintf("Heap buffer %s on address %s. Check indices used for dynamically allocated arrays.\n",
-          heapErrorType,
-          addrpatt.exec(contents[1])));
-        stack_trace(contents);
-      }
-      else if(/LeakSanitizer:/.test(contents[1])) { // memory leak
-        self._write("Memory leaks occurred:");
-        stack_trace(contents);
-      }
-      else if(/heap-use-after-free /.test(contents[1])) { // use after free
-        self._write(sprintf("Using address %s after it has been freed.\n",
-          addrpatt.exec(contents[1])));
-        stack_trace(contents);
+      for (var key2 in err.misc) {
+        maybe_log('printing outer misc: ' + key2);
+        to_print.push(key2.replace(/_/g, " ") + ': ' + err.misc[key2]);
       }
-      else if(/double-free /.test(contents[1])) { // double free
-        self._write(sprintf("Attempting to free address %s, which has already been freed.\n",
-          addrpatt.exec(contents[1])));
-        stack_trace(contents);
-      } else if(/attempting free on/.test(contents[1])) { // free, not malloc
-        self._write(sprintf("Attempting to free address %s, which not been malloc'd.\n",
-          addrpatt.exec(contents[1])));
-        stack_trace(contents);
-      }
-      else { // else print usual message
-        _.each(contents, function(line) {
-          self._write(line + "\n");
-        });
+      //maybe_log('done iterating over everything; to_print has ' + to_print.length + ' items');
+      //to_print.push(err.raw_message);
+      for (var j = 0; j < to_print.length; j++){
+        self._write(to_print[j] + '\n');
+        //console.log(to_print[j]);
       }
     }
 
     socket.register_callback("io", function(io) {
       if(io.type == "stdout") {
         ind = io.message.indexOf("\n");
-        if(ind > -1) {
+        if (ind > -1) {
           spl = io.message.split("\n");
           self._write(self.stdout);
-          while(spl.length>1) { self._write(spl.shift() + "\n"); }
+          while (spl.length>1) { self._write(spl.shift() + "\n"); }
           self.stdout = spl[0];
         }
-        else
+        else {
           self.stdout += io.message;
+        }
       }
       else if(io.type == "stderr") {
         ind = io.message.indexOf("\n");
-        if(ind > -1) {
-          if(!self.asan_parse)
+        if (ind > -1) {
+          if (!self.asan_parse) {
             self._write(self.stderr);
-          else
+          } else {
             io.message = self.stderr + io.message;
+          }
           spl = io.message.split("\n");
-          while(spl.length>1) {
-            if(!self.asan_parse && /^=+$/.test(spl[0])) {
+          while (spl.length>1) {
+            if (!self.asan_parse && /^=+$/.test(spl[0])) {
```

Comment:
Now that the ASAN output has been separated from stderr can you get rid of these tests on self.asan_parse?